### ISSUE TITLE
f_println, f_printAt, f_simpleprint bug

### ISSUE THREAD

**Issue by Chromowolf** [armoha/euddraft#28]
I'm reading the source code of `StringBuffer`. 
`f_println`, `f_printAt`, `f_simpleprint` have the following bug:
main.eps:
```js
const me = getuserplayerid();
function afterTriggerExec() {
	const cp = getcurpl();
	setcurpl(me);
	
	printAt(1, "Always display");

	if(getgametick() > 100) {
		printAt(0, "game tick > 100");
	}
	setcurpl(cp);
}
```
Everything goes normally:
![image](https://user-images.githubusercontent.com/37732663/113404043-2520eb00-93da-11eb-93d8-2598bef07ce1.png)

But if I change the order of the two `printAt`:
```js
const me = getuserplayerid();
function afterTriggerExec() {
	const cp = getcurpl();
	setcurpl(me);
	
	if(getgametick() > 100) {
		printAt(0, "game tick > 100");
	}
	
	printAt(1, "Always display");
	
	setcurpl(cp);
}
```
Result: "game tick > 100" and "Always display" are displayed at the same time (when `getgametick() > 100`):
![image](https://user-images.githubusercontent.com/37732663/113404648-14bd4000-93db-11eb-8936-ac537aaa3593.png)

All three functions `f_println`, `f_printAt`, `f_simpleprint` behave exactly the same. As long as "one function is called within a sub-environment, one function is called outside the environment afterwards", the bug occurs.
I think there may be some bug in `GetGlobalStringBuffer()`

**Comment by armoha** [comment]
This is known bug; initialization of local StringBuffer happens on first usage (`GlobalStringBuffer` is local in some expect that it is declared only when there is any usage.) So init triggers which run once is located before `printAt(0, "game tick > 100");`. It can be fixed by intializing lazily created EUDObject on game start just like `IsUserCP()` condition or etc does.

**Comment by Chromowolf** [comment]
So you mean I could write the following code in `FixPrintBug.eps`
```js
function onPluginStart() {
	const cp = getcurpl();
	setcurpl(getuserplayerid());
	GetGlobalStringBuffer();
	setcurpl(cp);
}
```
and make `FixPrintBug.eps` the first plugin in `main.edd`. This can fix all the print bugs?

**Comment by armoha** [comment]
Yes it does. It makes GlobalStringBuffer for map which does not use it though. Btw you don't need any CurrentPlayer triggers.

What I've thinking is making `GetMapStringAddr` to const function. STR section address is fixed in SC:R and so are every string. This will eliminate any need for init trigger.

**Comment by Chromowolf** [comment]
Thx. Looking forward to the fix

**Comment by armoha** [comment]
Fixed in armoha/eudplib@537080a75f0814175b9dbe0882ff4b5741d85710

No longer have to call `GetGlobalStringBuffer()` in `onPluginStart`

**Comment by armoha** [comment]
This is completely fixed with armoha/euddraft#112 in next euddraft version, `GetMapStringAddr` becomes constant function, no need to run any trigger at all to initialize `StringBuffer`.
